### ISSUE TITLE
[MWRAPPER-93] Distribution sha256 checksum not validated if the zip file was downloaded previously

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-wrapper#226]
**[Jorge Solórzano](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jorsol)** opened **[MWRAPPER-93](https://issues.apache.org/jira/browse/MWRAPPER-93?redirect=false)** and commented

If I make a first run without **distributionSha256Sum**, the Maven distribution will be downloaded without any checksum, this is the normal behavior.

But if I later add the **distributionSha256Sum** to the maven-wrapper.properties file, having downloaded previously the distribution, the checksum is not verified, I consider this a bug since even if the distribution is already downloaded and unpacked it could contain a compromised download.

The options **alwaysUnpack** and **alwaysDownload** triggers the verification and provides an extra layer of security, but normally the local zip should be verified always if the **distributionSha256Sum** is set.

---

**Affects:** 3.2.0

**Comment by jira-importer** [comment]
**[Jorge Solórzano](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jorsol)** commented

This might be closed as it looks like it is the requested behavior, and the **alwaysUnpack** or **alwaysDownload** should be used. See: https://github.com/apache/maven-wrapper/pull/58#issuecomment-1265742206

I still believe that as a "security" feature, this check should always be done (https://github.com/apache/maven-wrapper/pull/58#issuecomment-1405007910), but if there is no need to check every single time, then this issue can be closed.

**Comment by jira-importer** [comment]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** commented

[Jorge Solórzano](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jorsol) so this is fixed by MWRAPPER-75 ?

**Comment by jira-importer** [comment]
**[Jorge Solórzano](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jorsol)** commented

[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas) no, MWRAPPER-75 introduced that feature, the issue here is related to this comment:
https://github.com/apache/maven-wrapper/pull/58#issuecomment-1265742206

 

The checksum is validated only on download or unpack, but not every time the script runs, which might be fine depending on who you ask.

In other words, if the checksum should be validated on every run of the {**}mvnw{**}, or only when a download/unpack is done? If it's the latter, then this issue can be closed as Won't Fix.

**Comment by jira-importer** [comment]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** commented

On every run... then distro is already unpacked, no? So on what should checksum validation run? On downloaded distro? If so, what would guarantee that unpacked content really comes from distro?

**Comment by jira-importer** [comment]
**[Jorge Solórzano](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jorsol)** commented

Suppose that you already downloaded the maven distribution without the **distributionSha256Sum** set, this could be a "malicious" download, then you add the **distributionSha256Sum** but it will not check that the downloaded distribution is the correct one.

Again, this could be solved by using **alwaysUnpack** or **alwaysDownload** to ensure the validation is done and it's running the correct distribution, so under that premise, this issue can be closed as Won't Fix.
